samza-api/src/main/java/org/apache/samza/table/descriptors/CachingTableDescriptor.java

```diff
@@ -158,6 +130,34 @@ public TableSpec getTableSpec() {
     return this;
   }
 
+  @Override
+  public String getProviderFactoryClassName() {
+    return PROVIDER_FACTORY_CLASS_NAME;
+  }
+
+  @Override
+  protected void generateConfig(Config jobConfig, Map<String, String> tableConfig) {
```

Comment:
What's `tableConfig` in the method params?

Reply:
It's now refactored away